spec/linter-erlc-spec.js

```diff
@@ -0,0 +1,68 @@
+'use babel';
+
+// eslint-disable-next-line no-unused-vars
+import { it, fit, wait, beforeEach, afterEach } from 'jasmine-fix';
+import * as path from 'path';
+import linter from '../lib/init';
+
+const goodPath = path.join(__dirname, 'fixtures', 'good.erl');
+const badPath = path.join(__dirname, 'fixtures', 'bad.erl');
+const emptyPath = path.join(__dirname, 'fixtures', 'empty.erl');
+
+describe('The ERLC provider for Linter', () => {
+  const lint = linter.provideLinter().lint;
+
+  beforeEach(async () => {
+    await atom.packages.activatePackage('linter-erlc');
+  });
+
+  it('should be in the packages list', () =>
+    expect(atom.packages.isPackageLoaded('linter-erlc')).toBe(true),
+  );
+
+  it('should be an active package', () =>
+    expect(atom.packages.isPackageActive('linter-erlc')).toBe(true),
+  );
+
+  it('finds nothing wrong with a valid file', async () => {
+    const editor = await atom.workspace.open(goodPath);
+    const messages = await lint(editor);
+    expect(messages.length).toBe(0);
+  });
+
+  describe('shows errors in a file with issues', () => {
+    let editor = null;
+
+    beforeEach(async () => {
+      editor = await atom.workspace.open(badPath);
+    });
+
+    it('verifies the first message', async () => {
+      const expected = 'function unknown/0 undefined';
+
+      const messages = await lint(editor);
+      expect(messages[0].type).toBe('Error');
+      expect(messages[0].text).toBe(expected);
+      expect(messages[0].filePath).toBe(badPath);
+      expect(messages[0].range).toEqual([[1, 0], [1, 21]]);
+    });
+  });
+
+  describe('shows errors in an empty file', () => {
+    let editor = null;
+
+    beforeEach(async () => {
+      editor = await atom.workspace.open(emptyPath);
+    });
+
+    it('verifies the first message', async () => {
+      const expected = 'no module definition';
+
+      const messages = await lint(editor);
+      expect(messages[0].type).toBe('Error');
+      expect(messages[0].text).toBe(expected);
+      expect(messages[0].filePath).toBe(emptyPath);
+      expect(messages[0].range).toEqual([[0, 0], [0, 0]]);
```

Comment:
When I use `generateRange(textEditor, 0)` it generates the same range array ([[0,0],[0,0]]). This may be because the the whole file is empty, including the first line. Or maybe that's because this linter is still using the v1 and this behaviour has changed in v2?

In the editor, it does add the red circle on the first line and the error message is displayed when the line is hovered.
![screenshot from 2017-08-23 13-44-54](https://user-images.githubusercontent.com/7062224/29627837-cff4f944-8809-11e7-8bb2-c7d5b94db7f5.png)